### ISSUE TITLE
error on search

### ISSUE THREAD

**Issue by eatingcakeagain** [PhiTux/DailyTxT#9]
hello,

I have recently installed this docker image and entered ~30 entries. I wanted to test the search function with a word I know I entered, but I get the following error pop up in the lower right corner of the page... any support would be appreciated as I currently like the look and feel of this container and would like to use it more.

LINK: https://i.imgur.com/9vVZ85y.png

![chrome_3tLNccbfJg](https://user-images.githubusercontent.com/100134450/154952177-22fa31e7-ae01-43bd-8635-a29d356d854a.png)
 
"Invalid token. Registration and / or authentication required"

many thanks

**Comment by PhiTux** [comment]
Hi,

thanks for the message! This should obviously not happen.
Normally this message should only appear, when you are actually signed out and the authentication failed...

1. Is the rest of the program still working normally? Can you save new text or select another day? -> So, are you sure, you're signed in?
2. Does this error only appear on a specific searchword, or is the search not working at all?

**Comment by PhiTux** [comment]
And perhaps 3.: Could you give some example to rebuild this problem?

**Comment by eatingcakeagain** [comment]
Hi - thanks for getting back to me so quickly.
to answer your 3 comments/questions:

1. _Is the rest of the program still working normally? Can you save new text or select another day? -> So, are you sure, you're signed in?_   ------> As far as I can tell the rest is working properly; I have been able to export/backup the data as a zip, then recreated the container and imported it fresh again - this all still came with the same error though (sadly). Definitely signed in and able to create new items on various days.

2. _Does this error only appear on a specific searchword, or is the search not working at all?_   ------> I have tried with a handful of words that I have put in to the system and they all come up with the error.

3. _Could you give some example to rebuild this problem?_    ------> I am running this on an unraid system, downloaded the container from there... the odd thing was that it seemed to work one time after inputting 2 or 3 items in for last week , but have since added ~30 and since then the error comes up in the search - sorry there is no other way I can think of to recreate or rebuild the problem - unless I have installed incorrectly, but I took the "standard" settings to create the container.

**Comment by PhiTux** [comment]
Okay, that's really strange...

Just to be sure: Let's say you select today (02/21/2022) and there is the following text inside: 
"Today is a great day, because I am eating cake again."

This text is saved (green border) and visible even after switching back/forth one day and now you search for "day", then you should see a result "a great day, because I am"
And this is not the case?


And there is now way to install incorrectly thanks to docker

**Comment by eatingcakeagain** [comment]
Sadly I get the error message still - I even took your excellent example txt (I see what you did there)  :-) went on a day and back again and the error is there still.
![image](https://user-images.githubusercontent.com/100134450/154967656-e5399c1a-4a93-4f54-9e06-d5058318b148.png)

Search from an empty day (just in case that makes a difference...)
![image](https://user-images.githubusercontent.com/100134450/154968136-d98e04a1-d5b8-49a8-85c6-61004cf40b63.png)

**Comment by eatingcakeagain** [comment]
Ok . might it be worth me creating a "new" dailytxt container, and copy-pasting things in from the data export, and searching as I go... that might highlight an cause for the error? If it works then I can continue to use the new version and trash the old one... 
still doesn't get the rout cause though(?)

**Comment by PhiTux** [comment]
Yes, that sounds like a good idea to try to find the cause of the error - since you said, that it seemed to work, when you only had few entries made.
You could just start a new container with different data-path, so you start completely new. And then just fill this new one slowly and test the search regularly. 👍

**Comment by PhiTux** [comment]
Strange...
Did I understand it right, that you imported it now into a new User-Account in a new docker-container? Then it was perhaps some problem with your encryption-key/user-account!?

The history stays untouched from the search and there is no option to disable the history!

But so it's working now?

**Comment by eatingcakeagain** [comment]
Not exactly. What I did was....
- take a backup of the data I entered (creates a zip with various json files and any attachments - as you know)
- save that backup to my local machine
- completely remove/purge/delete the container and any images
- start a fresh instance of the container
- create a new account (therefore only one account exists)
- inspect the various json files from the backup and extract the relevant points/latest version from the history
- copy/paste the "clean" data from the json to the new dailytxt instance (i.e. I did not use the import function here)
- during this copy/paste I was checking search terms as I went along and there was no issues - all worked as expected

So yes, it could have been something to do with the encryption-key/user-account.
Long story short - it is working. thanks for the help and discussion above.

I have an idea for an small feature - where should I add this in GitHub? (sorry for the noob question)

**Comment by PhiTux** [comment]
Okay, I'm really pretty sure, that this was some error with the user-account / the JWT. I can not really imagine what the problem was exactly...
From this point of information I think that this was not a program error, since the search is something that can not produce such an error, as long as the rest of dailytxt is working normal...

But good, that it works now!

If you have a new feature, please just open a new issue :)

**Comment by PhiTux** [comment]
By coincidence, I had the exact same error today. 
And I'm 99% sure I know where it's coming from. It's just a minor programming bug - was neither a problem with your logs/saves nor with you user account/authentication.
Really sorry for that!

Will hopefully publish a fixed version within the next days :)

**Comment by eatingcakeagain** [comment]
Good to hear you hopefully have found the problem - It has not occurred to me since I did the clean install.

**Comment by PhiTux** [comment]
Yes, but it can reoccur at any time under a specific circumstance ;)
The problem lies within empty logs, that are not really empty. When you formerly saved something for a day and then delete it again, then this day is in some special state (intentional), which is obviously not correctly treated by the search.
